### PR TITLE
Workflow Update: abort all updates if workflow is terminated or timed out

### PR DESCRIPTION
**Pull request by alexshtin**
## What changed?

Abort all updates if workflow is terminated or timed out.

## Why?

For better DX. W/o this PR updates will just timeout.

## How did you test it?

Need to add tests.

## Potential risks

No risks.

## Documentation

No.

## Is hotfix candidate?

No.

### CODE REVIEW

File: service/history/api/workflow_lease.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ type workflowLease struct {
 type UpdateWorkflowAction struct {
 	Noop               bool
 	CreateWorkflowTask bool
+	AbortUpdates       bool
```
Comment (yycptt) [inline service/history/api/workflow_lease.go:47]: it feels like a different type of post actions compared to the other two... better to add some comments here...

File: service/history/ndc/workflow_resetter.go [hunk 2]
```diff
@@ -156,6 +156,8 @@ func (r *workflowResetterImpl) ResetWorkflow(
 			return err
 		}
 
+		currentWorkflow.GetContext().UpdateRegistry(ctx, nil).Abort(update.AbortReasonWorkflowCompleted)
```
Comment (yycptt) [inline service/history/ndc/workflow_resetter.go:159]: we haven't persisted the reset at this point right? That's done on L235.

File: service/history/workflow/update/abort_reason.go [hunk 3]
```diff
@@ -45,8 +44,6 @@ func (r AbortReason) Error() error {
 		return registryClearedErr
 	case AbortReasonWorkflowCompleted:
 		return consts.ErrWorkflowCompleted
```
Comment (yycptt) [inline service/history/workflow/update/abort_reason.go:46]: Probably should be renamed to ErrWorkflowClosed

Comment (alexshtin) [reply service/history/workflow/update/abort_reason.go:46]: Agree, I will do in separate PR.

**Review by yycptt — APPROVED** [review]

**Pull request merged**